pkg/operator/k8sutil/deployment.go

```diff
@@ -73,6 +90,40 @@ func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deploym
 		}
 		if d.Status.ObservedGeneration != original.Status.ObservedGeneration && d.Status.UpdatedReplicas > 0 && d.Status.ReadyReplicas > 0 {
 			logger.Infof("finished waiting for updated deployment %s", d.Name)
+
+			// Now we check if we can go to the next daemon
+			actionToCheck := "continue"
+			for _, daemon := range daemonsToCheckList {
+				if strings.Contains(deployment.Name, daemon) {
+					logger.Infof("checking if deployment %s update can continue", deployment.Name)
+					err := upgrade.OkToStopOrContinue(context, namespace, deployment.Name, daemon, actionToCheck, clusterInfo.CephVersion)
+					if err != nil {
+						return nil, fmt.Errorf("failed to check if deployment %s can be updated. %+v", deployment.Name, err)
+					}
+					if daemon == "mon" {
+						if clusterInfo.CephVersion.IsAtLeastNautilus() {
+							v, err := upgrade.GetCephMonVersion(context)
+							if err != nil {
+								return nil, fmt.Errorf("failed to get ceph mon version. %+v", err)
+							}
+
+							if v.IsAtLeastNautilus() {
+								versions, err := upgrade.GetCephVersions(context)
+								if err != nil {
+									return nil, fmt.Errorf("failed to get ceph daemons versions. %+v", err)
+								}
+								if len(versions.Mon) == 1 {
+									// If len is one, this clearly indicates that all the mons are running the same version
+									// We are doing this because 'ceph version' might return the Ceph version that a majority of mons have but not all of them
+									// so instead of trying to active msgr2 when mons are not ready, we activate it when we believe that's the right time
+									upgrade.EnableMessenger2(context, namespace)
```

Comment:
We can add anything that needs to be enabled for Nautilus here.

Reply:
The 1.0 tentative upgrade guide (#2965) mentions running this command. Should we run it here? And do we need to only run these commands once instead of every time the operator runs the orchestration?
```
ceph osd require-osd-release nautilus
```